devito/types/array.py

```diff
@@ -46,15 +49,26 @@ class Array(ArrayBasic):
         The halo region of the object.
     padding : iterable of 2-tuples, optional
         The padding region of the object.
+    space : str, optional
+        The memory space. Allowed values: 'default', 'remote', 'mapped',
+        'uniform'.  Defaults to 'default'. The 'default' value means the Array
+        is allocated in the underlying platform's default memory space. For
+        example, this will be the thread virtual memory space if running on a
+        CPU, or the global memory if running on a GPU. 'remote' is the dual of
+        'default' -- if running on a GPU, a 'remote' Array will be allocated on
+        the host in the virtual memory of the driving thread. 'mapped' means
+        the Array is accessible by all compute elements of the underlying node,
+        though proper synchronization is needed (i.e., will be introduced by
+        the Devito compiler) to ensure data coherence. 'uniform', instead,
+        means the Array will be used as if the underlying node had uniform
+        memory address space.
     scope : str, optional
-        Control memory allocation. Allowed values: 'heap', 'stack'. Defaults to 'heap'.
+        The scope in the given memory space. Allowed values: 'heap', 'stack'.
+        Defaults to 'heap'. This may not have an impact on certain platforms.
     sharing : str, optional
         Control data sharing. Allowed values: 'shared', 'local'. Defaults to 'shared'.
```

Comment:
This boat has sailed, but I feel 'local' is ambiguous; just because data is local doesn't mean it cannot be accessed 'remotely'. 'private' has a clearer intent and would be consistent with other programming models.

Reply:
noted. Will change